### ISSUE TITLE
Devise Master returns 200 on user create but Devise 3.5.2 returns 201

### ISSUE THREAD

**Issue by adnguyen1989** [plataformatec/devise#3943]
I have a Rails-based API. For a previous project (Rails 4.2.2, Ruby 2.2.2, Devise 3.5.2), I normally get 201 with the full User resource.

```
SQL (1.2ms)  INSERT INTO "users" ("email", "encrypted_password", "first_name", "last_name", "ui_language", "original_language", "target_language", "open_id", "created_at", "updated_at", "confirmation_token", "confirmation_sent_at") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)  [["email", "test15@gmail.com"], ["encrypted_password", "$2a$10$Fg1TOz97G1LCRlBx1yLhLeXHs2K.ZTgn/k9wi4okkHoZTQotBKbOa"], ["first_name", "anh"], ["last_name", "nguyen"], ["ui_language", "hi"], ["original_language", "hi"], ["target_language", "hi"], ["open_id", "-2ueX7V2hQQX51s12zqT"], ["created_at", "2016-02-10 13:14:12.848088"], ["updated_at", "2016-02-10 13:14:12.848088"], ["confirmation_token", "T6K-MVx2Hx_SxtPAp-P6"], ["confirmation_sent_at", "2016-02-10 13:14:13.081561"]]
  Rendered devise/mailer/confirmation_instructions.html.erb (4.8ms)

Devise::Mailer#confirmation_instructions: processed outbound mail in 280.1ms

Sent mail to test15@gmail.com (19.6ms)
Date: Wed, 10 Feb 2016 20:14:13 +0700
From: please-change-me-at-config-initializers-devise@example.com
Reply-To: please-change-me-at-config-initializers-devise@example.com
To: test15@gmail.com
Message-ID: <56bb37a55e029_131eb3fc5486ce13c29384@Anhs-MacBook-Pro.local.mail>
Subject: Confirmation instructions
Mime-Version: 1.0
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>Welcome test15@gmail.com!</p>

<p>You can confirm your account email through the link below:</p>

<p><a href="http://localhost:3000/users/confirmation?confirmation_token=T6K-MVx2Hx_SxtPAp-P6">Confirm my account</a></p>

   (0.8ms)  commit transaction
   (0.1ms)  begin transaction
  User Exists (0.1ms)  SELECT  1 AS one FROM "users" WHERE "users"."open_id" = ? LIMIT 1  [["open_id", "dJFaZGxMv8jrRyvWczth"]]
  SQL (0.4ms)  UPDATE "users" SET "last_sign_in_at" = ?, "current_sign_in_at" = ?, "last_sign_in_ip" = ?, "current_sign_in_ip" = ?, "sign_in_count" = ?, "open_id" = ?, "updated_at" = ? WHERE "users"."id" = ?  [["last_sign_in_at", "2016-02-10 13:14:13.400264"], ["current_sign_in_at", "2016-02-10 13:14:13.400264"], ["last_sign_in_ip", "127.0.0.1"], ["current_sign_in_ip", "127.0.0.1"], ["sign_in_count", 1], ["open_id", "dJFaZGxMv8jrRyvWczth"], ["updated_at", "2016-02-10 13:14:13.402382"], ["id", 1]]
   (0.8ms)  commit transaction
Completed 201 Created in 686ms (Views: 7.5ms | ActiveRecord: 5.2ms)
```

However on the latest project (Rails 5 beta 2, Ruby 2.3.0, Devise master), I get 200 with blank response.

```
SQL (0.9ms)  INSERT INTO "users" ("email", "encrypted_password", "confirmation_token", "confirmation_sent_at", "created_at", "updated_at", "first_name", "last_name", "open_id") VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9) RETURNING "id"  [["email", "test9@gmail.com"], ["encrypted_password", "$2a$10$GZnQLSKhoWRRopUlI6OSy.7glelY6qk.D6x5WruuHx53hOTZR4mVC"], ["confirmation_token", "Zeb-SuiTU3mekzusipuh"], ["confirmation_sent_at", 2016-02-10 12:39:31 UTC], ["created_at", 2016-02-10 12:39:31 UTC], ["updated_at", 2016-02-10 12:39:31 UTC], ["first_name", "anh"], ["last_name", "nguyen"], ["open_id", "cW9AS-MTLx2w9zfycSL-"]]
  Rendered /Users/jackof1trade/.rvm/gems/ruby-2.3.0/bundler/gems/devise-7c75ebe672f1/app/views/devise/mailer/confirmation_instructions.html.erb (8.2ms)
Devise::Mailer#confirmation_instructions: processed outbound mail in 223.2ms
Sent mail to test9@gmail.com (13.7ms)
Date: Wed, 10 Feb 2016 19:39:31 +0700
From: please-change-me-at-config-initializers-devise@example.com
Reply-To: please-change-me-at-config-initializers-devise@example.com
To: test9@gmail.com
Message-ID: <56bb2f83b47f6_d9d73fc34c8a347452999@Anhs-MacBook-Pro.local.mail>
Subject: Confirmation instructions
Mime-Version: 1.0
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>Welcome test9@gmail.com!</p>

<p>You can confirm your account email through the link below:</p>

<p><a href="http://localhost:3000/users/confirmation?confirmation_token=Zeb-SuiTU3mekzusipuh">Confirm my account</a></p>

   (0.6ms)  COMMIT
Completed 200 OK in 619ms (Views: 0.2ms | ActiveRecord: 11.8ms)
```

Since a resource was created successfully the best practice is for 201 to be returned with at least the resource ID. Any idea why this might have changed?

**Comment by adnguyen1989** [comment]
I found some other inconsistencies in response code. I'm not sure if it's Devise or Rails 5

**Comment by lucasmazza** [comment]
Can you please provide a sample application that reproduces the error?

**Comment by adnguyen1989** [comment]
@lucasmazza 

Please try the failing RSpec tests in this sample (Rails 5 beta 2, Ruby 2.3.0, Devise 4.0.0RC1): https://github.com/adnguyen1989/rails_api_template/tree/user_management

A working Rails 4.2.2, Ruby 2.2.2, Devise 3.5.2 sample:
https://github.com/adnguyen1989/yomiapi

**Comment by lucasmazza** [comment]
This is an incompatibility with the `responders` gem and `ActionController::API`, as the responders default rendering relies too much on the controller implementation.

There is a proposal fix at plataformatec/responders#146.
